ambry-api/src/main/java/com.github.ambry/clustermap/ClusterSpectator.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import org.apache.helix.api.listeners.InstanceConfigChangeListener;
+import org.apache.helix.api.listeners.LiveInstanceChangeListener;
+
+
+/**
+ * {@link ClusterSpectator} adds itself as a spectator of a cluster and registers some cluster change listeners.
+ */
+public interface ClusterSpectator extends InstanceConfigChangeListener, LiveInstanceChangeListener {
+
+  /**
+   * Register with helix cluster as a spectator.
+   */
+  void spectate() throws Exception;
```

Comment:
I checked `VcrHelixClusterSpector`, it seems this method is more like a `start()` method that can be done in the constructor. For other implementations in the future, I wonder if this method is necessary. Maybe we can simplify this just like `ClusterChangeHandler` in `HelixClusterManager`.